consul_v1.go

```diff
@@ -15,12 +15,13 @@ var errFuncNotImplemented = fmt.Errorf("function is not implemented")
 // namespaces.
 func FuncMapConsulV1() template.FuncMap {
 	return template.FuncMap{
-		"service":   v1ServiceFunc,
-		"connect":   v1ConnectFunc,
-		"services":  v1ServicesFunc,
-		"keys":      v1KVListFunc,
-		"key":       v1KVGetFunc,
-		"keyExists": v1KVExistsFunc,
+		"service":      v1ServiceFunc,
+		"connect":      v1ConnectFunc,
+		"services":     v1ServicesFunc,
+		"keys":         v1KVListFunc,
+		"key":          v1KVGetFunc,
+		"keyExists":    v1KVExistsFunc,
+		"keyExistsGet": v1KVExistsGetFunc,
```

Comment:
In general, not sure about what to call this method, but since it's kind of an evolution of key exists and key get, I just went with `keyExistsGet`. Totally down for changing it if anyone has a better idea!

Reply:
Yeah, I think it's okay... I can't think of anything else either

Reply:
I'll leave it for now then, should be trivial to update in the future if any inspiration hits us!